src/App.tsx

```diff
@@ -1,34 +1,118 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+type Props = {
+  query: string;
+  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleSelectChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  preparedTodos: Todo[];
+  currentTodo: Todo;
+  onClose: () => void;
+};
+
+export const App: React.FC<Props> = () => {
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectValue, setSelectValue] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  function getPreparedTodo(todosForQuery: Todo[], newQuery: string) {
+    let preparedTodos = [...todosForQuery];
+
+    const queryIgnore = newQuery.toLowerCase().trim();
+
+    if (newQuery) {
+      preparedTodos = preparedTodos.filter(todo =>
+        todo.title.toLowerCase().includes(queryIgnore),
+      );
+    }
+
+    return preparedTodos;
+  }
+
+  const [query, setQuery] = useState('');
```

Comment:
Move this state to the state blocks above 
<img width="622" alt="image" src="https://github.com/user-attachments/assets/a07c905c-064b-4b06-abf5-9f51623dbc83">